modules/index/src/main/java/org/apache/ignite/internal/index/ChangeIndexStatusTask.java

```diff
@@ -149,28 +158,33 @@ CompletableFuture<Void> start() {
         }
 
         try {
-            return supplyAsync(() -> awaitActivateForCatalogVersionOfIndexCreation()
-                    .thenCompose(unused -> ensureThatLocalNodeStillPrimaryReplica())
-                    .thenCompose(unused -> inBusyLocks(logicalTopologyService::logicalTopologyOnLeader))
-                    .thenComposeAsync(this::awaitFinishRwTxsBeforeCatalogVersionOfIndexCreation, executor)
-                    .thenComposeAsync(unused -> switchIndexToBuildingStatus(), executor), executor)
-                    .thenCompose(Function.identity())
-                    .handle((unused, throwable) -> {
+            LOG.info("Starting task to change index status. Index: {}", indexDescriptor);
+
+            return awaitCatalogVersionActivation()
```

Comment:
Please bring back the use of `supplyAsync` so that in the thread that will call the code we do not start, for example, going to the network and so on.

Reply:
Can you please clarify? What's the problem of going to the network, if we wait for the response asynchronously?

Reply:
If you look at who calls this code, you can find that this happens on the metastore thread, which I want to release as soon as possible, if the futures are already completed, then the pipeline will be executed immediately, and we don’t know what’s inside it, maybe there are calculations that will take some time.

Perhaps this looks like premature optimization or worrying from scratch, but I think we need to slightly change the way we use listeners on the metastore thread.

Reply:
Fixed